### PR TITLE
Add sippfmt utility

### PR DESCRIPTION
**Pull request by vodik**
Contribute the `sippfmt` script I wrote. Still needs some documentation and probably needs some tweaking as I used it to pull out stuff internally that we though was useless - like copypasta comments that stopped making sense as they got replicated around... probably not for the general public's taste.

But its a start and I'd like some feedback

### CODE REVIEW

File: pysipp/xml/cleaner.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+def main():
+    parser = argparse.ArgumentParser(description='Process some sipp scripts')
+    parser.add_argument('filename')
+    args = parser.parse_args()
+
+    doc = process_document(args.filename)
+    xml = doc.toprettyxml(indent='  ', encoding='ISO-8859-1')
+
+    with open(args.filename, 'w') as fp:
```
Comment (tgoodlet) [inline pysipp/xml/cleaner.py]: Maybe by default it should just write to stdout?
That way a user doesn't mutate a file without redirecting it to the same file?

**Comment by tgoodlet** [comment]
I'm fine with bringing it in.
I just had one comment about mutating the input file.

File: pysipp/xml/cleaner.py (hunk 1 above)
Comment (vodik) [inline pysipp/xml/cleaner.py]: Yeah, agreed

**Comment by tgoodlet** [comment]
@vodik yeah since it's so simple just add an *Formatter Util* section to the readme which shows the basic usage.

File: pysipp/xml/cleaner.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    # delete the last seperator
+    if new_scen.childNodes and isinstance(new_scen.childNodes[-1], minidom.Newline):
+        del new_scen.childNodes[-1]
+
+    doc.appendChild(new_scen)
+    return doc
+
+
+def main():
```
Comment (wdoekes) [inline pysipp/xml/cleaner.py]: This lacks some documentation about what it does.

It looks like it reformats scenario files, but "Process some sipp scripts" doesn't quite cover that.

File: pysipp/xml/minidom.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        writer.write(newl)
+
+
+def createSeperator(self):
+    c = Newline()
+    c.ownerDocument = self
+    return c
+
+
+Document.createSeperator = createSeperator
```
Comment (wdoekes) [inline pysipp/xml/minidom.py]: There's "a-rat" in separator ;)

(That 'a' is audible in romanic/germanic languages, causing the typo to stand out.)

File: pysipp/xml/cleaner.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                continue
+            elif comment.startswith("Handle 200 OK for CANCEL"):
+                in_comment = True
+                continue
+            elif comment.startswith("200 Ok to the CANCEL request OR INVITE request"):
+                in_comment = True
+                continue
+            elif comment.startswith("***************************"):
+                in_comment = True
+                continue
```
Comment (wdoekes) [inline pysipp/xml/cleaner.py]: This huge list of comment.startswith is awkward.

First of all, startswith takes an optional tuple:
```
if comment.startswith((
        'Handle 200 OK for CANCEL',
        '200 Ok to the CANCEL')):
    in_comment=True
```

But secondly, what are we discerning between? I can't tell from the code why these phrases are special.

File: pysipp/xml/minidom.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+class AttributeSorter(object):
+    def __init__(self, obj, *args):
+        self.obj = obj
+
+    def __lt__(self, other):
+        if self.obj in ('request', 'response'):
+            return True
+        elif other.obj in ('request', 'response'):
+            return False
+        return locale.strcoll(self.obj, other.obj) < 0
```
Comment (wdoekes) [inline pysipp/xml/minidom.py]: Do you really want locale based collation? Shouldn't that just be `self.obj < other.obj` ?

File: pysipp/xml/cleaner.py (hunk 4 above)
Comment (vodik) [inline pysipp/xml/cleaner.py]: I didn't intend to commit this actually. They where special boilerplate comments that ended up in most of our internal sipp scripts - comments that didn't ever get updated so where always out-of-date.

Comment (vodik) [inline pysipp/xml/cleaner.py]: Oh, also didn't know about the tuple support. Thanks.

**Comment by vodik** [comment]
Sorry it took me so long to get back to this - I forgot. Addressed feedback, added some documentation, and rebased onto master.

File: setup.py [hunk 6]
```diff
@@ -38,6 +38,11 @@
     packages=['pysipp'],
```
Comment (wdoekes) [inline setup.py]:
```-    packages=['pysipp'],
+    packages=['pysipp', 'pysipp.cli'],
```

Comment (vodik) [inline setup.py]: Oh right, good catch.

**Comment by vodik** [comment]
Indeed you're right. Not sure how to format that, but I'll put some thought into it. I think I'm going to merge this and open an issue for it then. Dealt with the packaging issue, good catch.

**Comment by tgoodlet** [comment]
@vodik looks good, although I've seen it before ;)

Any chance we can get some simple docs for newcomers either in the README or wiki section?

**Pull request merged**